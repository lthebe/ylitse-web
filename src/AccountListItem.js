import React from 'react';
import PropTypes from 'prop-types';
import { TableCell, TableRow } from 'material-ui/Table';
import IconButton from 'material-ui/IconButton';
import UpdateIcon from 'material-ui-icons/Update';
import DeleteIcon from 'material-ui-icons/Delete';

const AccountListItem = ({ role, username, email }) => (
    <TableRow>
        <TableCell>{role}</TableCell>
        <TableCell>{username}</TableCell>
        <TableCell>{email}</TableCell>
        <TableCell>
            <IconButton><UpdateIcon onClick={() => {}} /></IconButton>
            <IconButton onClick={() => {}}><DeleteIcon /></IconButton>
        </TableCell>
    </TableRow>
);

AccountListItem.propTypes = {
    role: PropTypes.string.isRequired,
    username: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
};

export default AccountListItem;
