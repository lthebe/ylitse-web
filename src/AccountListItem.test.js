import React from 'react';
import { shallow } from 'enzyme';

import AccountListItem from './AccountListItem';

describe('AccountListItem', () => {
    test('renders correctly', () => {
        const account = {
            username: 'admin',
            role: 'admin',
            email: 'admin@admin.com',
        };
        const wrapper = shallow(<AccountListItem account={account} />);

        expect(wrapper).toMatchSnapshot();
    });
});
