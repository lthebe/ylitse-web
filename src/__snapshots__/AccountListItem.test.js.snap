// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountListItem renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(AccountListItem)
    account={
      Object {
        "email": "admin@admin.com",
        "role": "admin",
        "username": "admin",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "account": Object {
        "email": "admin@admin.com",
        "role": "admin",
        "username": "admin",
      },
      "classes": Object {
        "firstCell": "AccountListItem-firstCell-1",
        "icon": "AccountListItem-icon-4",
        "lastCell": "AccountListItem-lastCell-2",
        "lastIconButton": "AccountListItem-lastIconButton-3",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "account": Object {
          "email": "admin@admin.com",
          "role": "admin",
          "username": "admin",
        },
        "classes": Object {
          "firstCell": "AccountListItem-firstCell-1",
          "icon": "AccountListItem-icon-4",
          "lastCell": "AccountListItem-lastCell-2",
          "lastIconButton": "AccountListItem-lastIconButton-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
